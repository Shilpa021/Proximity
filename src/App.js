import React, { useState, useEffect } from "react";
import Home from "./components/Home/Home.js";
import { Container } from "@material-ui/core";
import { AppBar, Toolbar, Typography } from "@material-ui/core";
import "./App.scss";

const App = (props) => {
  //give an initial state so that the data won't be undefined at start
  const [aqiData, setAqiData] = useState([]);

  useEffect(() => {
    const websocket = new WebSocket("ws://city-ws.herokuapp.com/"); // Using websockets to connect to the server

    websocket.onopen = () => {
      console.log("connected"); // consoles when websocket is opened
    };

     /* Fetches data */
    websocket.onmessage = function (event) {
      const json = JSON.parse(event.data);
      setAqiData(json);
    };

    return () => {
      websocket.close();
    };
  }, []);

  return (
    <div className="page">

      <AppBar>
        <Toolbar>
          <Typography variant="h5">{props.header}</Typography>
        </Toolbar>
      </AppBar>

      <Container maxWidth="lg">
        <Home aqiData={aqiData} />
      </Container>
      
    </div>
  );
};

/**
 * Default values for passed properties
 *
 * @type {object}
 * @property {string} header='Air Quality Monitor' - The default value for header
 */
App.defaultProps = {
  header: "Air Quality Monitor"
};

export default App;
