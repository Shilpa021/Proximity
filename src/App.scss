.App {
  text-align: center;
}

.page {
  padding: 84px;
  font-family: circular, sans-serif;
  background-color: powderblue;
  background: url("https://images.unsplash.com/photo-1606994868513-d480e1a1cab0?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1770&q=80");
  header {
    margin: 0 auto;
    position: static;
    background-color: teal;
    color: aliceblue;
  }
}

.MuiToolbar-regular {
  justify-content: center;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

@media only screen and (max-width: 600px) {
  .page {
    padding: 48px;
  }
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
