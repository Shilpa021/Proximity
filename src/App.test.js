import App from "./App";
import renderer from "react-test-renderer";
import Enzyme, { mount } from "enzyme";

import Adapter from "@wojtekmaj/enzyme-adapter-react-17"; // added for Adapter

Enzyme.configure({ adapter: new Adapter() });

test("renders without crashing", () => {
  mount(<App />);
});

test("should render 1 header", () => {
  const component = mount(<App />);
  expect(component.find("header").length).toBe(1);
});

test("should render 1 h5 tag", () => {
  const component = mount(<App />);
  expect(component.find("h5").length).toBe(1);
});

test("should render 1 h5 tag with text Air Quality Monitor", () => {
  const component = mount(<App />);
  expect(component.find("h5").text()).toBe("Air Quality Monitor");
});

test("should render 1 Home component", () => {
  const component = mount(<App />);
  expect(component.find("Home").length).toBe(1);
});

test("should render 1 Button with text Compare AQIs", () => {
  const component = mount(<App />);
  expect(component.find("button").length).toBe(1);
  expect(component.find("button span").text()).toBe("Compare AQIs");
});

test("should render 1 table, tbody and thead ", () => {
  const component = mount(<App />);
  expect(component.find("table").length).toBe(1);
  expect(component.find("thead").length).toBe(1);
  expect(component.find("tbody").length).toBe(1);
});

test("renders correctly", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
