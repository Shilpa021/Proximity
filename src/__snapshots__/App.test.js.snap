// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="page"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5"
      >
        Air Quality Monitor
      </h5>
    </div>
  </header>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      className="MuiBox-root MuiBox-root-29"
    >
      <div
        className="button"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Compare AQIs
          </span>
        </button>
      </div>
      <div
        className="MuiBox-root MuiBox-root-31"
      >
        <div
          className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          <table
            aria-label="simple table"
            className="MuiTable-root makeStyles-table-30"
            role={null}
          >
            <thead
              className="MuiTableHead-root"
              role={null}
            >
              <tr
                className="MuiTableRow-root MuiTableRow-head"
                role={null}
              >
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-paddingCheckbox"
                  scope="col"
                />
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                  scope="col"
                >
                  City
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                  scope="col"
                >
                  Current AQI
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
                  scope="col"
                >
                  Last Updated
                </th>
              </tr>
            </thead>
            <tbody
              className="MuiTableBody-root"
              role={null}
            />
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
